### PR TITLE
Builds: Eliminate insecure docker login

### PR DESCRIPTION
**Pull request by and-rewsmith**
Replacing manual docker login with the service connection approach. There is one occurrence of manual docker login I could not replace as it happens through an ssh connection.
https://github.com/Azure/iotedge/blob/master/builds/misc/templates/mqtt-perf-setup.yaml#L44

### CODE REVIEW

File: builds/templates/install-dotnet2.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+steps:
+  - task: UseDotNet@2
+    displayName: Install .NET Core sdk
+    inputs:
+      packageType: sdk
+      version: 2.2.207
```
Comment (and-rewsmith) [inline builds/templates/install-dotnet2.yaml:6]: I am restoring this file as we need it for the codesign.

File: builds/misc/addons-publish.yaml [hunk 2]
```diff
@@ -20,13 +20,19 @@ jobs:
           - checkout: self
             clean: true
             fetchDepth: 100
-        
-          - task: Bash@3
-            displayName: Log into Registries
+
+          # Both docker logins needed for if we need to test this job. In this case images should go to edgebuilds.
+          - task: Docker@2
+            displayName: Docker login
```
Comment (yophilav) [inline builds/misc/addons-publish.yaml:26]: Could you update the display name to be more indicative of what CR is being log into please?

File: builds/misc/images-release.yaml [hunk 3]
```diff
@@ -15,10 +15,13 @@ jobs:
...
-            docker login '$(build.registry.address)' --username '$(build.registry.user)' --password '$(build.registry.password)'
-            docker login '$(registry.address)' --username '$(registry.user)' --password '$(registry.password)'
-        displayName: 'Docker Login'
+      - task: Docker@2
+        displayName: Docker login
+        inputs:
+          command: login
+          containerRegistry: iotedge-release-acr,
+      # Dotnet 2 needed for codesign
+      - template: ../templates/install-dotnet2.yaml
```
[the remark this answers is not in the document]
Comment (yophilav) [reply builds/misc/images-release.yaml:24]: I thought the newer version of VM image has this dependency comes pre-installed.
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-README.md#net-core-sdk

[the remark this answers is not in the document]
Comment (and-rewsmith) [reply builds/misc/images-release.yaml:29]: The agent the release job picked up failed without this template. I think it is best to be explicit about dotnet 2 being required.

**Review by yophilav — APPROVED** [review]
Please run the pipelines at least once to make sure they are all good before merging. Thank you!

**Pull request merged**